rubin_scheduler/scheduler/surveys/field_survey.py

```diff
@@ -172,8 +174,15 @@ def __init__(
                     obs["scheduler_note"] = self.scheduler_note
                     self.observations.append(obs)
 
-        # Let's just make this an array for ease of use
-        self.observations = np.concatenate(self.observations)
+        # Let's just make this an array for ease of use if not already
+        if not isinstance(self.observations, ObservationArray):
+            self.observations = np.concatenate(self.observations)
+        # and backfill "band" if it was set by the user in "filter"
+        # Need to add "band" here if it wasn't populated
+        missing_band = np.where(self.observations["band"] == "")
+        band = np.char.rstrip(self.observations["filter"][missing_band], chars="_0123456789")
```

Comment:
This looks like hard-coding filter naming conventions that aren't documented anywhere. Should there be a utility function for converting `filter` to `band` so it's not a burried auto magical thing?

Reply:
Not a bad idea.

Reply:
Ok, so because of how I needed to use this here, I added a utility that operates on the entire ObservationArray. I think this should only be needed for cases where sequences are set up in the configuration for FieldSurvey (and this should only be the case in configurations for the FBS which are not properly updated for the band-filter swap). 
The best documentation I could find about the filters for LSST was in fact in the code for obs_lsst, at https://github.com/lsst/obs_lsst/blob/main/python/lsst/obs/lsst/filters.py 
.. but you can see in the ConsDB for example "band" and "physical filter" names for many observations. 
Examples include r_03 == r.  For Auxtel, where gratings as well as filter can be used, this  particular conversion doesn't really work.